xs2a-adapter-aspsp-registry/src/main/java/de/adorsys/xs2a/adapter/registry/LuceneAspspRepositoryFactory.java

```diff
@@ -0,0 +1,111 @@
+package de.adorsys.xs2a.adapter.registry;
+
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import de.adorsys.xs2a.adapter.registry.exception.RegistryIOException;
+import de.adorsys.xs2a.adapter.registry.mapper.AspspMapper;
+import de.adorsys.xs2a.adapter.service.model.Aspsp;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+import org.mapstruct.factory.Mappers;
+
+import java.io.*;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+public class LuceneAspspRepositoryFactory {
+    private static final String CSV_ASPSP_ADAPTER_CONFIG_FILE_PATH_PROPERTY = "csv.aspsp.adapter.config.file.path";
+    private static final String DEFAULT_CSV_ASPSP_ADAPTER_CONFIG_FILE = "aspsp-adapter-config.csv";
+    private static final String DEFAULT_LUCENE_DIR_PATH = "lucene";
+
+    private final AspspMapper aspspMapper = Mappers.getMapper(AspspMapper.class);
+
+    public LuceneAspspRepository newLuceneAspspRepository() {
+        try {
+            return newLuceneAspspRepositoryInternal();
+        } catch (IOException e) {
+            throw new RegistryIOException(e);
+        }
+    }
+
+    private LuceneAspspRepository newLuceneAspspRepositoryInternal() throws IOException {
+        byte[] csv = getCsvFileAsByteArray();
+
+        MessageDigest messageDigest = null;
+        try {
+            messageDigest = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
```

Comment:
Currently, `RestExceptionHandler` doesn't know to handle `RuntimeException`, so it will be handled as `Exception`. Is it correct behavior for our case?

Reply:
Or maybe it will be better to throw the concrete exception for this very case, what do you think?

Reply:
Every implementation of Java is required to support MD5 according to JavaDoc on MessageDigest class. So it's very unlikely that this exception will ever be thrown. And even if it will be that will happen on start-up not after a request.